tsdb/chunkenc/chunk.go

```diff
@@ -52,6 +52,24 @@ func IsValidEncoding(e Encoding) bool {
 	return e == EncXOR || e == EncHistogram || e == EncFloatHistogram
 }
 
+const (
+	// MaxBytesPerXORChunk is a soft cap for the size of an XOR chunk. It is a soft cap in that a final
+	// sample may be appended that exceeds the cap but no further samples will be appended to that chunk.
+	// This results in an effective hard cap of 1024 bytes (1005 bytes already in a chunk followed by a
+	// sample that takes the maximum size of 19 bytes).
+	MaxBytesPerXORChunk = 1005
```

Comment:
Since this is an exposed method saying "MaxBytes", I would suggest setting this to 1024 since that is our actual max that we intend to have (and easier for consumers of this variable to reason about). The comment makes sense on why it was 1005, but we could move that comment and logic to where we are using it. i.e., in head_append.go, we can put similar comment and check for `len(c.chunk.Bytes()) > chunkenc.MaxBytesPerXORChunk-19` instead.

Reply:
Sounds good, will do!